gtars-io/src/bigbed.rs

```diff
@@ -0,0 +1,131 @@
+use std::collections::HashMap;
+use std::path::Path;
+
+use gtars_core::models::RegionSet;
+use gtars_core::utils::get_chrom_sizes;
+
+use bigtools::beddata::BedParserStreamingIterator;
+use bigtools::{BedEntry, BigBedWrite as BigBedWriter};
+use tokio::runtime;
+
+pub trait BigBedWrite {
+    ///
+    /// Write data as bigBed (binary version of bed file)
+    ///
+    /// # Arguments
+    /// - out_path: the path to the bigbed file which should be created
+    /// - chrom_size: the path to chrom sizes file
+    ///
+    fn write_bigbed<P: AsRef<Path>>(
+        &self,
+        out_path: P,
+        chrom_size: P,
+    ) -> Result<(), std::io::Error>;
+}
+
+impl BigBedWrite for RegionSet {
+    fn write_bigbed<P: AsRef<Path>>(
+        &self,
+        out_path: P,
+        chrom_size: P,
+    ) -> Result<(), std::io::Error> {
+        let out_path = out_path.as_ref();
+        if out_path.exists() {
+            println!("Bed file already exists. Overwriting existing file")
+        }
+
+        if let Some(parent) = out_path.parent() {
+            std::fs::create_dir_all(parent)?;
+        }
+        let chrom_sizes: HashMap<String, u32> = get_chrom_sizes(chrom_size);
+
+        let mut warnings_count: i32 = 0;
+        let region_vector = self.regions.iter().map(|i| {
+            // This if it is removing regions that are not in chrom sizes file.
+            if !chrom_sizes.contains_key(&i.chr) {
+                eprintln!(
+                    "Warning:: Chromosome is not found in Chrom sizes file. Chr: '{}'",
+                    i.chr
+                );
+                if warnings_count > 40 {
+                    panic!("Incorrect chrom sizes provided. Unable to create bigBed file!");
```

Comment:
If we are doing this restructuring, can we try to return error here instead of panic?